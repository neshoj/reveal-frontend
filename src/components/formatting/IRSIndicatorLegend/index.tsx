import React from 'react';
import { Col, Row, Table } from 'reactstrap';
import {
  IndicatorThresholdItem,
  IndicatorThresholds,
  indicatorThresholdsIRS,
} from '../../../configs/settings';
import { percentage } from '../../../helpers/utils';

interface Props {
  indicatorThresholds: IndicatorThresholds;
}

const defaultProps: Props = {
  indicatorThresholds: indicatorThresholdsIRS,
};

const IRSIndicatorLegend = (props: Props) => {
  const { indicatorThresholds } = props;
  const indicatorItems = Object.values(indicatorThresholds);

  return (
    <div className="card mt-5 mb-5">
      <div className="card-header">Conditional formatting rules</div>
      <div className="card-body">
        <Row>
          <Col sm="12" md={{ size: 4, offset: 4 }}>
            <Table className="text-center">
              <tbody>
                {indicatorItems
                  .sort((a: IndicatorThresholdItem, b: IndicatorThresholdItem) =>
                    a.value > b.value ? 1 : -1
                  )
                  .map((item, index) => {
                    return (
                      <tr key={index} style={{ background: item.color }}>
                        <td>{item.name}</td>
                        <td>
                          {index === 0
                            ? `< ${percentage(item.value)}`
                            : `${percentage(indicatorItems[index - 1].value)} - ${percentage(
                                item.value
                              )}`}
                        </td>
                      </tr>
                    );
                  })}
              </tbody>
            </Table>
          </Col>
        </Row>
      </div>
    </div>
  );
};

IRSIndicatorLegend.defaultProps = defaultProps;

export default IRSIndicatorLegend;
