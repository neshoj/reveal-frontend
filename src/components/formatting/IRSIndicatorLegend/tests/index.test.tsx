import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import IRSIndicatorLegend from '..';
import { indicatorThresholdsIRS } from '../../../../configs/settings';

describe('/components/formatting/IRSIndicatorLegend', () => {
  it('renders correctly', () => {
    const wrapper = mount(<IRSIndicatorLegend />);
    expect(wrapper.props()).toEqual({ indicatorThresholds: indicatorThresholdsIRS });
    expect(toJson(wrapper.find('.card-header'))).toMatchSnapshot('header');
    expect(toJson(wrapper.find('Table'))).toMatchSnapshot('indicator table');
    wrapper.unmount();
  });
});
