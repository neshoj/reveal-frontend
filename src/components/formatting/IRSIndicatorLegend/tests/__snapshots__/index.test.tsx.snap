// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/formatting/IRSIndicatorLegend renders correctly: header 1`] = `
<div
  className="card-header"
>
  Conditional formatting rules
</div>
`;

exports[`/components/formatting/IRSIndicatorLegend renders correctly: indicator table 1`] = `
<Table
  className="text-center"
  responsiveTag="div"
  tag="table"
>
  <table
    className="text-center table"
  >
    <tbody>
      <tr
        key="0"
        style={
          Object {
            "background": "#dddddd",
          }
        }
      >
        <td>
          Grey
        </td>
        <td>
          &lt; 20%
        </td>
      </tr>
      <tr
        key="1"
        style={
          Object {
            "background": "#FF4136",
          }
        }
      >
        <td>
          Red
        </td>
        <td>
          20% - 75%
        </td>
      </tr>
      <tr
        key="2"
        style={
          Object {
            "background": "#FFDC00",
          }
        }
      >
        <td>
          Yellow
        </td>
        <td>
          75% - 90%
        </td>
      </tr>
      <tr
        key="3"
        style={
          Object {
            "background": "#2ECC40",
          }
        }
      >
        <td>
          Green
        </td>
        <td>
          90% - 100%
        </td>
      </tr>
    </tbody>
  </table>
</Table>
`;
